tests/test_interface.rb

```diff
@@ -78,11 +80,12 @@ def interface_count
 
   # Helper to check for misc speed change disallowed error messages.
   def speed_change_disallowed(message)
-    pattern = Regexp.new('(port doesn t support this speed|' \
-                         'Changing interface speed is not permitted|' \
-                         'requested config change not allowed)')
+    patterns = %w(port doesn t support this speed
```

Comment:
Even with the union this can't be right -- you'll match against any single word in `patterns`. Please fix.